### PR TITLE
Add certificate in the correct configuration for HA use case

### PR DESCRIPTION
**Pull request by tdemaret**
### What does this PR do?
When keycloack start, he use the standalone-ha.xml configuration but in entrypoint the certificate is added in standalone.xml : 
/usr/lib/jvm/java/bin/java -D[Standalone] -server -Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true -Dorg.jboss.boot.log.file=/opt/jboss/keycloak/standalone/log/server.log -Dlogging.configuration=file:/opt/jboss/keycloak/standalone/configuration/logging.properties -jar /opt/jboss/keycloak/jboss-modules.jar -mp /opt/jboss/keycloak/modules org.jboss.as.standalone -Djboss.home.dir=/opt/jboss/keycloak -Djboss.server.base.dir=/opt/jboss/keycloak/standalone -Djboss.bind.address=10.128.5.254 -Djboss.bind.address.private=10.128.5.254 **-c=standalone-ha.xml** -Dkeycloak.migration.action=import -Dkeycloak.migration.provider=dir -Dkeycloak.migration.strategy=IGNORE_EXISTING -Dkeycloak.migration.dir=/scripts/ -Djboss.bind.address=0.0.0.0

### What issues does this PR fix or reference?
Fix issue when you deploy Che on Openshift for multi user use case with self sign certificate for the master nodes.

### CODE REVIEW

**Comment by benoitf** [comment]
cc @skabashnyuk

**Comment by skabashnyuk** [comment]
CC @tolusha

**Comment by slemeur** [comment]
@skabashnyuk : as this is new area for @tolusha do you mind giving a hand in the review process of the PR?

**Comment by skabashnyuk** [comment]
@slemeur I don't mind.  Keycloak configuration is a new area for me as well.  Not sure my expertise is valuable here.

**Comment by tdemaret** [comment]
It is not necessary to be an expert, just if you read the entrypoint and see the run command, you can see the script doesn't modify the correct file when you run in multi-user mode with self-sign certificate on master nodes.

**Comment by tolusha** [comment]
@slemeur @skabashnyuk 
I can't validate the PR

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
hello @tdemaret in order to be able to merge PR, Eclipse ECA needs to be signed (currently the check is failing)
you can click on the status link to get more info (SignedOff footer)

**Comment by benoitf** [comment]
@tolusha ci-build was not performed (it explained why can't be merged)

**Comment by benoitf** [comment]
ci-build

**Comment by tolusha** [comment]
ci-test

**Comment by tolusha** [comment]
@tdemaret
Hi. Unfortunately it isn't clear what do you want to fix. 
We haven't faced any issues with self sign certificate. Could you provide any steps how you installed Che on minishift with keycloack in HA mode?

**Comment by tdemaret** [comment]
Hi @tolusha ,

I deploy on openhift 3 with this documentation : https://github.com/eclipse/che/tree/master/deploy/openshift/templates
With this option mode : Deploy multi user Che with bundled Keycloak and Postgres (https)
and connect keycloak on Openshift Master OpenId Connect.

**Comment by tdemaret** [comment]
@tolusha  : If I am alone with this use case, I can close this PR, but if you take 2 minutes to read the keycloak entrypoint, you can just see : it is not the same file when keycloak start and the file for trust store.

**Comment by tolusha** [comment]
I am testing...

File: dockerfiles/keycloak/kc_realm_user.sh [hunk 1]
```diff
@@ -37,7 +37,8 @@ if [ "${CHE_SELF__SIGNED__CERT}" != "" ]; then
     echo "${CHE_SELF__SIGNED__CERT}" > /scripts/openshift.cer
     keytool -importcert -alias HOSTDOMAIN -keystore /scripts/openshift.jks -file /scripts/openshift.cer -storepass openshift -noprompt
     keytool -importkeystore -srckeystore $JAVA_HOME/jre/lib/security/cacerts -destkeystore /scripts/openshift.jks -srcstorepass changeit -deststorepass openshift
-    /opt/jboss/keycloak/bin/jboss-cli.sh --file=/scripts/cli/add_openshift_certificate.cli && rm -rf /opt/jboss/keycloak/standalone/configuration/standalone_xml_history
+    /opt/jboss/keycloak/bin/jboss-cli.sh --file=/scripts/cli/add_openshift_certificate.cli && rm -f /opt/jboss/keycloak/standalone/configuration/standalone_xml_history
```
Comment (tolusha) [inline dockerfiles/keycloak/kc_realm_user.sh:40]:
```suggestion
```

Comment (tolusha) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: I don't think we need this line anymore.

Comment (tdemaret) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: Yes, but I am not sure, if I have an issue if I remove this line.

Comment (tolusha) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: Could you try pls on your side?

Comment (tdemaret) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: Yes, it works for me without this line, but I am not sure if it is never used.

Comment (tolusha) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: @davidfestal 
WDYT

Comment (tdemaret) [reply dockerfiles/keycloak/kc_realm_user.sh:40]: I prefer to keep both and add a comment for the first line.

**Comment by davidfestal** [comment]
> I deploy on openhift 3 with this documentation :  https://github.com/eclipse/che/tree/master/deploy/openshift/templates
With this option mode : Deploy multi user Che with bundled Keycloak and Postgres (https)
and connect keycloak on Openshift Master OpenId Connect.

The recommended way to install Che on OpenShift 3 is `Chectl` that will use the operator-based installation method: https://www.eclipse.org/che/docs/che-7/installing-che-on-openshift-3-using-the-operator/

And if you look into the Che operator related code, you can see that the current configuration is **not**  the HA configuration, but the **standalone** one, as shown in line https://github.com/eclipse/che-operator/blob/master/pkg/deploy/deployment_keycloak.go#L65

See also at line https://github.com/eclipse/che-operator/blob/master/pkg/deploy/deployment_keycloak.go#L218

So I assume that, if any fix is necessary to have the [obsolete installation method](https://github.com/eclipse/che/tree/master/deploy/openshift/templates) to still work correctly, it would be more by setting the `standalone` configuration in the Keycloak start command line, probably by adding the following arguments:

```
-c standalone.xml
```

in line https://github.com/eclipse/che/blob/a33cc5fc9b83450e75730cbed7d28b3774f80e26/dockerfiles/keycloak/kc_realm_user.sh#L50

Then this would bring back consistency between all installation methods.

Hope it helps.

**Comment by tdemaret** [comment]
@davidfestal : Operator is a tech preview on Openhift 3 : https://docs.openshift.com/container-platform/3.11/install_config/installing-operator-framework.html

I can t use it, if I want to use a support.

**Comment by davidfestal** [comment]
> @davidfestal : Operator is a tech preview on Openhift 3 : https://docs.openshift.com/container-platform/3.11/install_config/installing-operator-framework.html
> 
> I can t use it, if I want to use Redhat support.

@tdemaret here we have to distinguish 2 things:
- The Operator Lifecycle Manager (OLM) component with the associated *OperatorHub* UI, which automates and simplifies the installation of operators. As you noted, this was a tech preview on OpenShift 3.11.
- The Operators *per se*, built through the Operator SDK, which consist of some K8S Custom Resource Definition objects + a controller to act on them. Such operators can *without any problem* be used on OpenShift 3.11. The only limitation on OS 3.11 is that they have to be installed *manually*, by applying the required YAML K8S definitions. This is exactly what the `chectl` tool mentioned in https://www.eclipse.org/che/docs/che-7/installing-che-on-openshift-3-using-the-operator/ does. It manually installs the Che operator in OpenShift, since automatic / graphical installation through OLM is not possible on OpenShift 3.11.

So to summarize the same Che operator is used for installing Che on Openshift 3.11 and 4.x. Only the installation of the operator itself is different.

Anyway, if you still want to install Che with the [old, unsupported installation method](https://github.com/eclipse/che/tree/master/deploy/openshift/templates), please try to update your PR to apply the change described in comment https://github.com/eclipse/che/pull/15568#issuecomment-585094969 : instead of adding the certificate in the *ha* configuration, we should fix the code to use the `standalone` configuration when starting. Thus it will be consistent with what the supported operator-based installation currently does.
